### PR TITLE
Initial commit Transactional Outbox pattern

### PR DESCRIPTION
**Pull request by cdennig**
Hi @ckittel,

here's the initial version of the transactional outbox implementation with Cosmos DB. I just copied the code to the repo without the history, because in the original one, there were also only three commits.

Best,
Christian

### CODE REVIEW

**Review by ckittel — APPROVED** [review]
@cdennig -- only really minor questions/observations below.

That said, nothing is blocking, so feel free to merge.  Awesome work.

File: Contacts.API/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+EXPOSE 80
+EXPOSE 443
+
+FROM mcr.microsoft.com/dotnet/sdk:5.0 AS build
+WORKDIR /src
+COPY ["Contacts.API/Contacts.API.csproj", "Contacts.API/"]
+RUN dotnet restore "Contacts.API/Contacts.API.csproj"
+COPY . .
+WORKDIR "/src/Contacts.API"
+RUN dotnet build "Contacts.API.csproj" -c Release -o /app/build
```
[the remark this answers is not in the document]
Comment (cdennig) [reply Contacts.API/Dockerfile:12]: Right, this cannot work. I'm having a look at it.

[the remark this answers is not in the document]
Comment (cdennig) [reply Contacts.API/Dockerfile:12]: I just tried out. Line 10 copies everything from the source directory into the container. That's why it works. It's from the template of the IDE, so I would not adjust it now. Anyway, found an issue with the `global.json` file which did not have a proper `rollForward` property set and therefore led to build errors with newer sdk images.

File: Contacts.Domain/Contact.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        }
+
+        public void SetDescription(string description)
+        {
+            Description = description;
+
+            if (IsNew) return;
+
+            AddEvent(new ContactDescriptionUpdatedEvent(Id, Description));
+            ModifiedAt = DateTimeOffset.Now;
```
[the remark this answers is not in the document]
Comment (cdennig) [reply Contacts.Domain/Contact.cs]: Right, this is best practices anyway, will replace all occurrences with `.UtcNow`.

File: Contacts.API/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (ckittel) [reply Contacts.API/Dockerfile:12]: oh, duh, yeah I totally missed line 10.  Yikes, it was right there the whole time.  _le sigh_

File: Contacts.Domain/Contact.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ckittel) [reply Contacts.Domain/Contact.cs]: thx!

**Pull request merged**